### PR TITLE
Docker and Github Actions

### PR DESCRIPTION
**Pull request by liamkeily**

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
```
Comment (ziadoz) [inline .github/workflows/continuous-integration.yml:12]: Is this env var used anywhere?

**Comment by ziadoz** [comment]
Looks good. I think we should be more explicit about version in the `composer.json` file too whilst we're here:

```
    "php": "^7.4 || ^8.0",
    "ext-SimpleXML": "^7.4 || ^8.0",
    "ext-json": "^7.4 || ^8.0",
    "ext-xmlreader": "^7.4 || ^8.0",
    "ext-xmlwriter": "^7.4 || ^8.0"
```

File: .gitignore [hunk 2]
```diff
@@ -7,6 +7,7 @@ vendor
 
 .phpunit.result.cache
 .php-cs-fixer.cache
+.php_cs.cache
```
Comment (maxakropolis) [inline .gitignore:10]: Should we have one of these files? Isn't it from an old version of PHP CS Fixer?

File: .php_cs.cache [hunk 3]
```diff
@@ -0,0 +1 @@
...hp":3638536871,"src\/Writer\/Entity\/Category.php":727697203,"src\/Writer\/Entity\/DeletedProduct.php":702757992,"src\/Writer\/Entity\/Product.php":1307386162,"src\/Writer\/Entity\/CustomAttribute.php":3446753471,"src\/Writer\/Entity\/Variant.php":3652802165,"src\/Writer\/Entity\/WriteableEntityInteface.php":1550672502,"src\/Writer\/Entity\/DeletedCategory.php":4044479396,"src\/Writer\/Entity\/DeletedAssignment.php":42814339,"src\/Writer\/Entity\/Assignment.php":2107866917,"src\/Writer\/Xml\/XmlWriter.php":4135224338,"src\/Writer\/Xml\/XmlFormatterException.php":3166338478,"src\/Writer\/Xml\/NotEmptyWriter.php":3281015189,"src\/Writer\/Xml\/XmlFormatter.php":4120015297,"src\/Writer\/EntityWriter\/CategoryXmlWriter.php":680262121,"src\/Writer\/EntityWriter\/AssignmentXmlWriter.php":1901939966,"src\/Writer\/EntityWriter\/DeletedProductXmlWriter.php":3432497426,"src\/Writer\/EntityWriter\/DeletedAssignmentXmlWriter.php":3859287862,"src\/Writer\/EntityWriter\/ProductXmlWriter.php":1462905887,"src\/Writer\/EntityWriter\/CustomAttributeWriter.php":2885219160,"src\/Writer\/EntityWriter\/DeletedCategoryXmlWriter.php":1014756056,"src\/Writer\/EntityWriter\/VariantXmlWriter.php":4125945097}}
```
Comment (ziadoz) [inline .php_cs.cache:1]: @liamkeily You've committed a cache file here.

Comment (weshooper) [reply .php_cs.cache:1]: @ziadoz see https://github.com/fusionspim/php-demandware-xml/pull/101

**Pull request merged**